src/main/java/FoodController.java

```diff
@@ -0,0 +1,101 @@
+import SearchIDs.SearchCollection;
+import fromIDs.FoodCollection;
+import hu.akarnokd.rxjava3.swing.SwingSchedulers;
+import io.reactivex.rxjava3.schedulers.Schedulers;
+
+import javax.inject.Inject;
+import javax.inject.Named;
+import javax.swing.*;
+import java.util.ArrayList;
+
+public class FoodController {
+    public FoodService service;
+    private final JLabel nameOfFood;
+    private final JLabel ingredients;
+    private final JLabel nutrients;
+    private final JLabel totals;
+    private final JLabel idsList;
+    private  ArrayList<TotalValue> totalValues = new ArrayList<>(0);
+    @Inject
+    public FoodController(FoodService foodService,
+                         @Named("chosenFoodName") JLabel nameOfFood,
+                          @Named("ingredientsList") JLabel ingredients,
+                          @Named("nutrientsList") JLabel nutrients,
+                          @Named("totalsList") JLabel totals,
+                          @Named("idsList") JLabel idsList){
+        this.service=foodService;
+        this.nameOfFood = nameOfFood;
+        this.ingredients = ingredients;
+        this.nutrients = nutrients;
+        this.totals = totals;
+        this.idsList = idsList;
+    }
+
+    public void updateInfo(String id){
+        service.getFoodInfo(id)
+                .subscribeOn(Schedulers.io())
+                .observeOn(SwingSchedulers.edt())
+                .subscribe(this::setInfo, Throwable::printStackTrace);
+    }
+    public void setInfo(FoodCollection foodCollection){
+        updateNameOfFood(foodCollection);
+        updateIngredients(foodCollection);
+        updateNutrients(foodCollection);
+        updateTotals();
+    }
+    public void updateNameOfFood(FoodCollection foodCollection){
+        nameOfFood.setText(foodCollection.getDescription());
+    }
+    public void updateIngredients(FoodCollection foodCollection){
+        ingredients.setText(foodCollection.getIngredients());
+    }
+    public void updateNutrients(FoodCollection foodCollection){
+        String list = "";
+        for (int i = 0; i < foodCollection.getFoodNutrients().size(); i++) {
+            String name = foodCollection.getFoodNutrients().get(i).getNutrient().getName();
+            Double amount = foodCollection.getFoodNutrients().get(i).getAmount();
+            boolean found = false;
+            for (int j = 0;  j< totalValues.size(); j++) {
+                if(totalValues.get(j).getName().equals(name)) {
+                    totalValues.get(j).addToAmount(amount);
+                    found = true;
+                }
+            }
```

Comment:
We learned a way of doing this this semester without using a for loop to search for the matching value.